assets/model_monitoring/components/src/action_analyzer/action_analyzer_metrics_calculation/run.py

```diff
@@ -0,0 +1,344 @@
+        }),
+        "A good example response would be:",
+        "## Example Task #3:",
+        json.dumps({
+            RATING: 4,
+        }),
+        "## Example Task #4",
+        json.dumps({
+            CONTEXT: "The Queen's Royal Castle is a well-known tourist attraction in the United Kingdom. It spans \
+                over 500 acres and contains extensive gardens and parks. The castle was built in the 15th century \
+                    and has been home to generations of royalty.",
+            PROMPT: "What are the main attractions of the Queen's Royal Castle?",
+            COMPLETION: "The main attractions of the Queen's Royal Castle are its expansive 500-acre grounds, \
+                extensive gardens, parks, and the historical castle itself, which dates back to the 15th century \
+                    and has housed generations of royalty.",
+        }),
+        "A good example response would be:",
+        "## Example Task #4:",
+        json.dumps({
+            RATING: 5,
+        }),
+        "User:",
+        "{input_samples}"
+    ]
+)
+
+
+def _query_relevance_scores(
+    turns: List[Tuple[str, str, str]],
+    template: str,
+    session: requests.Session,
+    endpoint_url: str,
+    token_manager: _APITokenManager,
+    model: str,
+    temperature: float,
+    top_p: float,
+    num_samples: int,
+    frequency_penalty: float,
+    presence_penalty: float,
+    max_tokens=3000,
+    stop: str = None
+) -> List[int]:
+
+    prompts = [template.replace("{input_samples", f"\n{json.dumps({'prompt': turn[0], 'completion': turn[1], 'context': turn[2]}, indent=4)}") for turn in turns]  # noqa: E501
+    print("prompts:", prompts)
+    ratings = []
+    for prompt in prompts:
+        request_data = {
+            "model": model,
+            "temperature": temperature,
+            "top_p": top_p,
+            "n": num_samples,
+            "max_tokens": max_tokens,
+            "frequency_penalty": frequency_penalty,
+            "presence_penalty": presence_penalty,
+            "messages": [
+                {"role": "user", "content": prompt}
+            ]
+        }
+
+        if stop:
+            request_data["stop"] = stop
+
+        response = {}
+        try:
+            response, time_taken = _request_api(
+                session=session,
+                endpoint_url=endpoint_url,
+                token_manager=token_manager,
+                **request_data,
+            )
+
+            # Append time taken to the line
+            response["response_time_sec"] = time_taken
+            print(response["samples"][0])
+            rating = json.loads(response["samples"][0])["rating"]
+            # print("===response===")
+            # print("rating=", rating, type(rating))
+            ratings.append(rating)
+        except Exception as e:  # noqa: B902
+            response["finish_reason"] = ["error"]
+            response["error"] = [str(e)]
+            raise e
```

Comment:
need to consider what will be the exception, and should we fail the job or let the job continue